### PR TITLE
Integration test for cluster reroute in

### PR DESCRIPTION
**Pull request by tbhanu-amzn**
### Description
Integration test for cluster reroute case
 
### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/kotlin/com/amazon/elasticsearch/replication/MultiClusterRestTestCase.kt [hunk 1]
```diff
@@ -334,6 +335,87 @@ abstract class MultiClusterRestTestCase : ESTestCase() {
         assertEquals(HttpStatus.SC_OK.toLong(), persistentConnectionResponse.statusLine.statusCode.toLong())
     }
 
+    protected fun getPrimaryNodeForShard(clusterName: String,indexname: String, shardNumber: String) :String {
+        val cluster = getNamedCluster(clusterName)
+        val persistentConnectionRequest = Request("GET", "/_cat/shards")
```
[the remark this answers is not in the document]
Comment (tbhanu-amzn) [reply src/test/kotlin/com/amazon/elasticsearch/replication/MultiClusterRestTestCase.kt:340]: done

File: src/test/kotlin/com/amazon/elasticsearch/replication/MultiClusterRestTestCase.kt [hunk 2]
```diff
@@ -334,6 +335,87 @@ abstract class MultiClusterRestTestCase : ESTestCase() {
...
+        val persistentConnectionRequest = Request("GET", "/$indexName/_search?pretty&q=*")
+
+        val persistentConnectionResponse = cluster.lowLevelClient.performRequest(persistentConnectionRequest)
+        val resp = EntityUtils.toString(persistentConnectionResponse.entity);
+        return resp
+    }
+
+    protected fun getNodesInCluster(clusterName: String) : List<String>{
+        val cluster = getNamedCluster(clusterName)
+        val persistentConnectionRequest = Request("GET", "/_cat/nodes")
```
[the remark this answers is not in the document]
Comment (tbhanu-amzn) [reply src/test/kotlin/com/amazon/elasticsearch/replication/MultiClusterRestTestCase.kt:384]: done

File: src/test/kotlin/com/amazon/elasticsearch/replication/integ/rest/ClusterRerouteFollowerIT.kt [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    private fun changeTemplate(clusterName: String) {
+        val cluster = getNamedCluster(clusterName)
+        val persistentConnectionRequest = Request("PUT", "_template/all")
+        val entityAsString = """
+                        {"template": "*", "settings": {"number_of_shards": 1, "number_of_replicas": 0}}""".trimMargin()
+
+        persistentConnectionRequest.entity = NStringEntity(entityAsString, ContentType.APPLICATION_JSON)
+        cluster.lowLevelClient.performRequest(persistentConnectionRequest)
+    }
+}
```
[the remark this answers is not in the document]
Comment (tbhanu-amzn) [reply src/test/kotlin/com/amazon/elasticsearch/replication/integ/rest/ClusterRerouteFollowerIT.kt:88]: done

File: src/test/kotlin/com/amazon/elasticsearch/replication/integ/rest/ClusterRerouteFollowerIT.kt [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            } catch (ex1 : Exception) {
+                logger.info("Changing template method is deprecated and throws an warning exception")
+            }
+            createConnectionBetweenClusters(FOLLOWER, LEADER)
+            val createIndexResponse = leaderClient.indices().create(CreateIndexRequest(leaderIndexName), RequestOptions.DEFAULT)
+            Assertions.assertThat(createIndexResponse.isAcknowledged).isTrue()
+            followerClient.startReplication(StartReplicationRequest("source", leaderIndexName, followerIndexName))
+            assertBusy ({
+                Assertions.assertThat(followerClient.indices().exists(GetIndexRequest(followerIndexName), RequestOptions.DEFAULT)).isEqualTo(true)
+            }, 1, TimeUnit.MINUTES)
```
[the remark this answers is not in the document]
Comment (tbhanu-amzn) [reply src/test/kotlin/com/amazon/elasticsearch/replication/integ/rest/ClusterRerouteFollowerIT.kt:42]: done

File: src/test/kotlin/com/amazon/elasticsearch/replication/integ/rest/ClusterRerouteFollowerIT.kt [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            insertDocToIndex(LEADER, "2", "dummy data 2",leaderIndexName)
+
+            //TODO: Fix this and uncomment this code
+            /*assertBusy ({
+                try {
+                    Assertions.assertThat(docs(FOLLOWER, followerIndexName)).contains("dummy data 2")
+                } catch (ex: Exception) {
+                    Assertions.assertThat(true).isEqualTo(false)
+                }
+            }, 1, TimeUnit.MINUTES)*/
```
[the remark this answers is not in the document]
Comment (tbhanu-amzn) [reply src/test/kotlin/com/amazon/elasticsearch/replication/integ/rest/ClusterRerouteFollowerIT.kt:73]: this is solved in the pr https://github.com/opensearch-project/cross-cluster-replication/pull/114

**Review by krishna-ggk — CHANGES_REQUESTED** [review]
Should we drop this since these are covered in diff PR?

**Comment by tbhanu-amzn** [comment]
This is already addressed in #114

**Pull request closed without merging**